src/HotChocolate/Core/src/Types.Scalars/UtcOffsetType.cs

```diff
@@ -0,0 +1,188 @@
+using System;
+using System.Collections.Generic;
+using HotChocolate.Language;
+
+namespace HotChocolate.Types
+{
+    /// <summary>
+    /// The `UtcOffset` scalar type represents a value of format ±hh:mm.
+    /// </summary>
+    public class UtcOffsetType : ScalarType<TimeSpan, StringValueNode>
+    {
+        private static readonly Dictionary<TimeSpan, string> _timeSpanToOffset =
+            new()
+            {
+                {new TimeSpan(-12,0,0), "-12:00"},
+                {new TimeSpan(-11,0,0), "-11:00"},
+                {new TimeSpan(-10,0,0), "-10:00"},
+                {new TimeSpan(-9,30,0), "-09:30"},
+                {new TimeSpan(-9,0,0), "-09:00"},
+                {new TimeSpan(-8,0,0), "-08:00"},
+                {new TimeSpan(-7,0,0), "-07:00"},
+                {new TimeSpan(-6,0,0), "-06:00"},
+                {new TimeSpan(-5,0,0), "-05:00"},
+                {new TimeSpan(-4,0,0), "-04:00"},
+                {new TimeSpan(-3,30,0), "-03:30"},
+                {new TimeSpan(-3,0,0), "-03:00"},
+                {new TimeSpan(-2,0,0), "-02:00"},
+                {new TimeSpan(-1,0,0), "-01:00"},
+                {new TimeSpan(0,0,0), "00:00"},
+                {new TimeSpan(1,0,0), "+01:00"},
+                {new TimeSpan(2,0,0), "+02:00"},
+                {new TimeSpan(3,0,0), "+03:00"},
+                {new TimeSpan(3,30,0), "+03:30"},
+                {new TimeSpan(4,0,0), "+04:00"},
+                {new TimeSpan(4,30,0), "+04:30"},
+                {new TimeSpan(5,0,0), "+05:00"},
+                {new TimeSpan(5,30,0), "+05:30"},
+                {new TimeSpan(5,45,0), "+05:45"},
+                {new TimeSpan(6,0,0), "+06:00"},
+                {new TimeSpan(6,30,0), "+06:30"},
+                {new TimeSpan(7,0,0), "+07:00"},
+                {new TimeSpan(8,0,0), "+08:00"},
+                {new TimeSpan(8,45,0), "+08:45"},
+                {new TimeSpan(9,0,0), "+09:00"},
+                {new TimeSpan(9,30,0), "+09:30"},
+                {new TimeSpan(10,0,0), "+10:00"},
+                {new TimeSpan(10,30,0), "+10:30"},
+                {new TimeSpan(11,0,0), "+11:00"},
+                {new TimeSpan(12,0,0), "+12:00"},
+                {new TimeSpan(12,45,0), "+12:45"},
+                {new TimeSpan(13,0,0), "+013:00"},
+                {new TimeSpan(14,0,0), "+14:00"},
+            };
+
+        private static readonly Dictionary<string, TimeSpan> _offsetToTimeSpan =
```

Comment:
Can we map these from the _timeSpanToOffset dictionary?
something like  
```csharp
  var dict =  _offsetToTimeSpan
        .Reverse()
        .ToDictionary(x => x.Value, x=> x.Key);

    //Then add the missing things
    dict["+00:00"] = ...
    dict["-00:00"] = ...
```

Reply:
The only issue is what about `+00:00` vs. `-00:00`? When the `TimeSpan` is the key it's just 0 which we can't set twice..